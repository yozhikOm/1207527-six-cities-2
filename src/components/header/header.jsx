import React from 'react';
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';

const Header = (props) => {
  const {isAuthorizationRequired, userInfo} = props;
  let route = `/login`;
  let linkTitle = `Sign in`;
  let userAvatarWrapper = <React.Fragment />;

  if (!isAuthorizationRequired) {
    route = `/favorites`;
    linkTitle = userInfo.name;
    userAvatarWrapper = <img src={`https://es31-server.appspot.com/six-cities` + userInfo.avatarUrl}/>;
  }

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to={`/`} className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41" />
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link to={route} className="header__nav-link header__nav-link--profile">
                  <div className="header__avatar-wrapper user__avatar-wrapper">
                    {userAvatarWrapper}
                  </div>
                  <span className="header__user-name user__name">{linkTitle}</span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  isAuthorizationRequired: PropTypes.bool,
  userInfo: PropTypes.shape({
    id: PropTypes.id,
    name: PropTypes.string,
    avatarUrl: PropTypes.string,
    isPro: PropTypes.bool
  }),
};

export {Header};
