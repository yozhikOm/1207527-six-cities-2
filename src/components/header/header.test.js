import React from 'react';
import {Header} from './header.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';

it(`Header компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  const mockUserInfo = {
    id: 2,
    name: `fakeUser`,
    avatarUrl: ``,
    isPro: true
  };

  renderer.render(<Header
    isAuthorizationRequired={true}
    userInfo={mockUserInfo}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
