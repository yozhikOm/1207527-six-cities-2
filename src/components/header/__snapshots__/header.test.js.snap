// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header компонент рендерится корректно 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <ForwardRef
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </ForwardRef>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <ForwardRef
              className="header__nav-link header__nav-link--profile"
              to="/login"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              >
                <React.Fragment />
              </div>
              <span
                className="header__user-name user__name"
              >
                Sign in
              </span>
            </ForwardRef>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
